### ISSUE TITLE
Not able to add webcontrol

### ISSUE THREAD

**Issue by MarcelakiMunaki** [home-assistant/home-assistant.github.io#35059]
### Feedback

First: thank you for making the integration i was waiting for long time! 👍 

Unfortunately i was not able to add my webcontrol. The device was discovered but when i tried to add it by just using host "webcontrol" it could not be added as their is no connection possible. When i'm using the ip-address the device is being added but their is an error message (in german: "Fehler beim Einrichten: Überprüfe die Protokolle")

This is what i find in the logfile:

`2024-10-04 05:36:56.718 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry 192.168.178.139 for wmspro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/wmspro/__init__.py", line 53, in async_setup_entry
    await dest.refresh()
  File "/usr/local/lib/python3.12/site-packages/wmspro/destination.py", line 93, in refresh
    self._actions[product["actionId"]]._update_params(product["value"])
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 6
`

My WebControl pro has the latest available firmware 12G, Bootloader 116, Container 11H. WebApp 3.8.1

### URL

https://www.home-assistant.io/integrations/wmspro/

### Version

2024.10.0

### Additional information

_No response_

**Comment by mback2k** [comment]
Thanks for the issue report. Can you please provide diagnostics for the added integration?

https://www.home-assistant.io/docs/configuration/troubleshooting/

Also, are you by any chance using the Tailscale addon?

**Comment by MarcelakiMunaki** [comment]
> Thanks for the issue report. Can you please provide diagnostics for the added integration?

This is what i'm getting:
`2024-10-04 08:02:27.350 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry 192.168.178.139 for wmspro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/wmspro/__init__.py", line 53, in async_setup_entry
    await dest.refresh()
  File "/usr/local/lib/python3.12/site-packages/wmspro/destination.py", line 93, in refresh
    self._actions[product["actionId"]]._update_params(product["value"])
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 6
2024-10-04 08:03:14.178 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry 192.168.178.139 for wmspro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/wmspro/__init__.py", line 53, in async_setup_entry
    await dest.refresh()
  File "/usr/local/lib/python3.12/site-packages/wmspro/destination.py", line 93, in refresh
    self._actions[product["actionId"]]._update_params(product["value"])
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 6
`

<img width="1105" alt="Bildschirmfoto 2024-10-04 um 08 03 22" src="https://github.com/user-attachments/assets/5f09ecf2-0908-4547-8a58-989de7d96ab9">

> Also, are you by any chance using the Tailscale addon?

no - not using the addon

**Comment by mback2k** [comment]
@MarcelakiMunaki this should be fixed with the upcoming HA patch release 2024.10.2.

Would you mind providing the integration diagnostics output once updated? I am curious to see what exactly caused this.

**Comment by MarcelakiMunaki** [comment]
> @MarcelakiMunaki this should be fixed with the upcoming HA patch release 2024.10.2.

awesome! my marquee is now showing up and able to control. Would be awesome to get control over the rest of my Warema Lamaxa L70 components in the future 👍 

> Would you mind providing the integration diagnostics output once updated? I am curious to see what exactly caused this.

Hope this is what you are looking for:

```
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.10.2",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.12.4",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/Berlin",
    "os_name": "Linux",
    "os_version": "6.6.31-haos-raspi",
    "supervisor": "2024.10.0",
    "host_os": "Home Assistant OS 13.1",
    "docker_version": "26.1.4",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "alarmo": {
      "documentation": "https://github.com/nielsfaber/alarmo",
      "version": "v1.10.5",
      "requirements": []
    },
    "multiscrape": {
      "documentation": "https://github.com/danieldotnl/ha-multiscrape",
      "version": "8.0.2",
      "requirements": [
        "lxml>=4.9.1",
        "beautifulsoup4>=4.12.2"
      ]
    },
    "dyson_local": {
      "documentation": "https://github.com/libdyson-wg/ha-dyson",
      "version": "1.4.2",
      "requirements": []
    },
    "waste_collection_schedule": {
      "documentation": "https://github.com/mampfes/hacs_waste_collection_schedule#readme",
      "version": "2.2.0",
      "requirements": [
        "icalendar",
        "recurring_ical_events",
        "icalevents!=0.1.28",
        "beautifulsoup4",
        "lxml",
        "pycryptodome"
      ]
    },
    "varta_storage": {
      "documentation": "https://www.home-assistant.io/integrations/varta_storage",
      "version": "v0.0.2",
      "requirements": [
        "vartastorage==0.2.5"
      ]
    },
    "chime_tts": {
      "documentation": "https://nimroddolev.github.io/chime_tts",
      "version": "v1.1.4",
      "requirements": [
        "pydub",
        "aiofiles"
      ]
    },
    "roborock": {
      "documentation": "https://github.com/humbertogontijo/homeassistant-roborock",
      "version": "1.0.13",
      "requirements": [
        "python-roborock==0.36.0",
        "ical==6.1.0",
        "dacite==1.8.0"
      ]
    },
    "wattpilot": {
      "documentation": "https://github.com/mk-maddin/wattpilot-HA",
      "version": "0.2.3",
      "requirements": [
        "wattpilot>=0.1",
        "pyyaml>=5.3.0",
        "importlib_metadata>=4.0.0",
        "aiofiles>=23.2.1",
        "packaging>=24.0"
      ]
    },
    "hacs": {
      "documentation": "https://hacs.xyz/docs/configuration/start",
      "version": "2.0.1",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    }
  },
  "integration_manifest": {
    "domain": "wmspro",
    "name": "WMS WebControl pro",
    "codeowners": [
      "mback2k"
    ],
    "config_flow": true,
    "dependencies": [],
    "dhcp": [
      {
        "macaddress": "0023D5*"
      },
      {
        "registered_devices": true
      }
    ],
    "documentation": "https://www.home-assistant.io/integrations/wmspro",
    "integration_type": "hub",
    "iot_class": "local_polling",
    "requirements": [
      "pywmspro==0.2.1"
    ],
    "is_built_in": true
  },
  "setup_times": {
    "null": {
      "setup": 0.00020314566791057587
    },
    "01J9ASJ6FSNCCC6NB7MVQYYT0X": {
      "wait_import_platforms": -12.403204895555973,
      "config_entry_setup": 16.284217040985823
    }
  },
  "data": {
    "host": "192.168.178.139",
    "config": {
      "command": "getConfiguration",
      "protocolVersion": "1.0.0",
      "destinations": [
        {
          "id": 45319,
          "animationType": 3,
          "names": [
            "Lamellen",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 0,
              "actionType": 0,
              "actionDescription": 2,
              "minValue": 0,
              "maxValue": 100
            },
            {
              "id": 16,
              "actionType": 6,
              "actionDescription": 12
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            },
            {
              "id": 23,
              "actionType": 7,
              "actionDescription": 12
            }
          ]
        },
        {
          "id": 108546,
          "animationType": 6,
          "names": [
            "Licht Traverse",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 0,
              "actionType": 0,
              "actionDescription": 8,
              "minValue": 0,
              "maxValue": 100
            },
            {
              "id": 17,
              "actionType": 6,
              "actionDescription": 12
            },
            {
              "id": 20,
              "actionType": 4,
              "actionDescription": 6
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        },
        {
          "id": 192974,
          "animationType": 6,
          "names": [
            "Licht Pfosten",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 0,
              "actionType": 0,
              "actionDescription": 8,
              "minValue": 0,
              "maxValue": 100
            },
            {
              "id": 17,
              "actionType": 6,
              "actionDescription": 12
            },
            {
              "id": 20,
              "actionType": 4,
              "actionDescription": 6
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        },
        {
          "id": 253507,
          "animationType": 5,
          "names": [
            "Heizstrahler 1",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 20,
              "actionType": 4,
              "actionDescription": 7
            },
            {
              "id": 21,
              "actionType": 5,
              "actionDescription": 10
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        },
        {
          "id": 264616,
          "animationType": 6,
          "names": [
            "Dimmen Heizstrahler 1",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 0,
              "actionType": 0,
              "actionDescription": 9,
              "minValue": 0,
              "maxValue": 100
            },
            {
              "id": 17,
              "actionType": 6,
              "actionDescription": 12
            },
            {
              "id": 20,
              "actionType": 4,
              "actionDescription": 7
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        },
        {
          "id": 390958,
          "animationType": 5,
          "names": [
            "Heizstrahler 2",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 20,
              "actionType": 4,
              "actionDescription": 7
            },
            {
              "id": 21,
              "actionType": 5,
              "actionDescription": 10
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        },
        {
          "id": 419551,
          "animationType": 6,
          "names": [
            "Dimmen Heizstrahler 2",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 0,
              "actionType": 0,
              "actionDescription": 9,
              "minValue": 0,
              "maxValue": 100
            },
            {
              "id": 17,
              "actionType": 6,
              "actionDescription": 12
            },
            {
              "id": 20,
              "actionType": 4,
              "actionDescription": 7
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        },
        {
          "id": 480806,
          "animationType": 1,
          "names": [
            "Senkrecht-Markise",
            "",
            "",
            ""
          ],
          "actions": [
            {
              "id": 0,
              "actionType": 0,
              "actionDescription": 0,
              "minValue": 0,
              "maxValue": 100
            },
            {
              "id": 16,
              "actionType": 6,
              "actionDescription": 12
            },
            {
              "id": 22,
              "actionType": 8,
              "actionDescription": 13
            }
          ]
        }
      ],
      "rooms": [
        {
          "id": 47875,
          "name": "Terrasse",
          "destinations": [
            45319,
            108546,
            192974,
            253507,
            264616,
            390958,
            419551,
            480806
          ],
          "scenes": []
        }
      ],
      "scenes": []
    },
    "dests": {
      "45319": {
        "id": 45319,
        "name": "Lamellen",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "0": {
            "id": 0,
            "actionType": "Percentage",
            "actionDescription": "SlatDrive",
            "attrs": {
              "minValue": 0,
              "maxValue": 100
            },
            "params": {
              "percentage": 100
            }
          },
          "16": {
            "id": 16,
            "actionType": "Stop",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {}
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          },
          "23": {
            "id": 23,
            "actionType": "Impulse",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {
              "rotation": 0
            }
          }
        },
        "animationType": "SlatRoof",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {
          "6": {
            "actionId": 6,
            "value": {
              "rotation": 0
            }
          }
        },
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 45319,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 0,
                    "value": {
                      "percentage": 100
                    }
                  },
                  {
                    "actionId": 6,
                    "value": {
                      "rotation": 0
                    }
                  },
                  {
                    "actionId": 23,
                    "value": {
                      "rotation": 0
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "108546": {
        "id": 108546,
        "name": "Licht Traverse",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "0": {
            "id": 0,
            "actionType": "Percentage",
            "actionDescription": "LightDimming",
            "attrs": {
              "minValue": 0,
              "maxValue": 100
            },
            "params": {}
          },
          "17": {
            "id": 17,
            "actionType": "Stop",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {}
          },
          "20": {
            "id": 20,
            "actionType": "Switch",
            "actionDescription": "LightSwitch",
            "attrs": {},
            "params": {}
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Dimmer",
        "drivingCause": "Unknown",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": null,
        "blocking": null,
        "status": {
          "protocolVersion": "1.0.0",
          "command": "getStatus",
          "errors": [
            327685
          ]
        }
      },
      "192974": {
        "id": 192974,
        "name": "Licht Pfosten",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "0": {
            "id": 0,
            "actionType": "Percentage",
            "actionDescription": "LightDimming",
            "attrs": {
              "minValue": 0,
              "maxValue": 100
            },
            "params": {
              "percentage": 0
            }
          },
          "17": {
            "id": 17,
            "actionType": "Stop",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {}
          },
          "20": {
            "id": 20,
            "actionType": "Switch",
            "actionDescription": "LightSwitch",
            "attrs": {},
            "params": {
              "onOffState": false
            }
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Dimmer",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 192974,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 0,
                    "value": {
                      "percentage": 0
                    }
                  },
                  {
                    "actionId": 20,
                    "value": {
                      "onOffState": false
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "253507": {
        "id": 253507,
        "name": "Heizstrahler 1",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "20": {
            "id": 20,
            "actionType": "Switch",
            "actionDescription": "LoadSwitch",
            "attrs": {},
            "params": {
              "onOffState": true
            }
          },
          "21": {
            "id": 21,
            "actionType": "Toggle",
            "actionDescription": "LightToggle",
            "attrs": {},
            "params": {
              "onOffState": true
            }
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Switch",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 253507,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 20,
                    "value": {
                      "onOffState": true
                    }
                  },
                  {
                    "actionId": 21,
                    "value": {
                      "onOffState": true
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "264616": {
        "id": 264616,
        "name": "Dimmen Heizstrahler 1",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "0": {
            "id": 0,
            "actionType": "Percentage",
            "actionDescription": "LoadDimming",
            "attrs": {
              "minValue": 0,
              "maxValue": 100
            },
            "params": {
              "percentage": 0
            }
          },
          "17": {
            "id": 17,
            "actionType": "Stop",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {}
          },
          "20": {
            "id": 20,
            "actionType": "Switch",
            "actionDescription": "LoadSwitch",
            "attrs": {},
            "params": {
              "onOffState": false
            }
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Dimmer",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 264616,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 0,
                    "value": {
                      "percentage": 0
                    }
                  },
                  {
                    "actionId": 20,
                    "value": {
                      "onOffState": false
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "390958": {
        "id": 390958,
        "name": "Heizstrahler 2",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "20": {
            "id": 20,
            "actionType": "Switch",
            "actionDescription": "LoadSwitch",
            "attrs": {},
            "params": {
              "onOffState": true
            }
          },
          "21": {
            "id": 21,
            "actionType": "Toggle",
            "actionDescription": "LightToggle",
            "attrs": {},
            "params": {
              "onOffState": true
            }
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Switch",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 390958,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 20,
                    "value": {
                      "onOffState": true
                    }
                  },
                  {
                    "actionId": 21,
                    "value": {
                      "onOffState": true
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "419551": {
        "id": 419551,
        "name": "Dimmen Heizstrahler 2",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "0": {
            "id": 0,
            "actionType": "Percentage",
            "actionDescription": "LoadDimming",
            "attrs": {
              "minValue": 0,
              "maxValue": 100
            },
            "params": {
              "percentage": 0
            }
          },
          "17": {
            "id": 17,
            "actionType": "Stop",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {}
          },
          "20": {
            "id": 20,
            "actionType": "Switch",
            "actionDescription": "LoadSwitch",
            "attrs": {},
            "params": {
              "onOffState": false
            }
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Dimmer",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 419551,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 0,
                    "value": {
                      "percentage": 0
                    }
                  },
                  {
                    "actionId": 20,
                    "value": {
                      "onOffState": false
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "480806": {
        "id": 480806,
        "name": "Senkrecht-Markise",
        "room": {
          "47875": "Terrasse"
        },
        "actions": {
          "0": {
            "id": 0,
            "actionType": "Percentage",
            "actionDescription": "AwningDrive",
            "attrs": {
              "minValue": 0,
              "maxValue": 100
            },
            "params": {
              "percentage": 0
            }
          },
          "16": {
            "id": 16,
            "actionType": "Stop",
            "actionDescription": "ManualCommand",
            "attrs": {},
            "params": {}
          },
          "22": {
            "id": 22,
            "actionType": "Identify",
            "actionDescription": "Identify",
            "attrs": {},
            "params": {}
          }
        },
        "animationType": "Awning",
        "drivingCause": "_None",
        "available": true,
        "unknownProducts": {},
        "heartbeatError": false,
        "blocking": false,
        "status": {
          "command": "getStatus",
          "protocolVersion": "1.0.0",
          "details": [
            {
              "destinationId": 480806,
              "data": {
                "drivingCause": 0,
                "heartbeatError": false,
                "blocking": false,
                "productData": [
                  {
                    "actionId": 0,
                    "value": {
                      "percentage": 0
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    },
    "rooms": {
      "47875": {
        "id": 47875,
        "name": "Terrasse",
        "destinations": {
          "45319": "Lamellen",
          "108546": "Licht Traverse",
          "192974": "Licht Pfosten",
          "253507": "Heizstrahler 1",
          "264616": "Dimmen Heizstrahler 1",
          "390958": "Heizstrahler 2",
          "419551": "Dimmen Heizstrahler 2",
          "480806": "Senkrecht-Markise"
        },
        "scenes": {}
      }
    },
    "scenes": {}
  }
}
```
